Source/BoundaryConditions/PEC_Insulator.cpp

```diff
@@ -0,0 +1,593 @@
+                    ? fbndry_lo[idim] == FieldBoundaryType::PECInsulator
+                    : fbndry_hi[idim] == FieldBoundaryType::PECInsulator );
+                if (isPEC_InsulatorBoundary) {
+                    isInsulatorBoundary = ( (iside == -1)
+                        ? is_insulator_lo[idim] == 1
+                        : is_insulator_hi[idim] == 1 );
+                }
+                if (isPEC_InsulatorBoundary) {
+                    // grid point ijk_vec is ig number of points pass the
+                    // domain boundary in direction, idim
+                    int const ig = ::get_cell_count_to_boundary(
+                        dom_lo, dom_hi, ijk_vec, is_nodal, idim, iside);
+
+#if (defined WARPX_DIM_XZ) || (defined WARPX_DIM_RZ)
+                    // For 2D : for icomp==1, (Fy in XZ, Ftheta in RZ),
+                    //          icomp=1 is not normal to x or z boundary
+                    //          The logic below ensures that the flags are set right for 2D
+                    is_normal_to_boundary = (icomp == (2*idim));
+#elif (defined WARPX_DIM_1D_Z)
+                    // For 1D : icomp=0 and icomp=1 (Fx and Fy are not normal to the z boundary)
+                    //          The logic below ensures that the flags are set right for 1D
+                    is_normal_to_boundary = (icomp == 2);
+#else
+                    is_normal_to_boundary = (icomp == idim);
+#endif
+
+                    if (ig == 0) {
+                        // Check if field is on the boundary
+                        if (is_nodal[idim] == 1) {
+                            OnBoundary = true;
+                        }
+                    } else if (ig > 0) {
+                        GuardCell = true;
+
+                        // Mirror location inside the domain by "ig" number of cells
+                        ijk_mirror[idim] = ( (iside == -1)
+                                        ? (dom_lo[idim] + ig - (1 - is_nodal[idim]))
+                                        : (dom_hi[idim] + 1 - ig));
+                        // Location twice as far in, for extrapolation
+                        ijk_mirrorp1[idim] = 2*ijk_mirror[idim] - ijk_vec[idim];
+
+                        // Check for components with even symmetry.
+                        // True for E_like and tangential, and B_like and normal
+                        if (E_like ^ is_normal_to_boundary) { sign *= -1._rt; }
+
+                        field_value = ( (iside == -1) ? field_lo[idim] : field_hi[idim] );
+                        set_field = ( (iside == -1) ? set_field_lo[idim]==1 : set_field_hi[idim]==1 );
+
+#if (defined WARPX_DIM_RZ)
+                        if (idim == 0 && iside == +1) {
+                            // Upper radial boundary
+                            amrex::Real const rguard = ijk_vec[idim] + 0.5_rt*(1._rt - is_nodal[idim]);
+                            if (icomp == 0) {
+                                // Add radial scale so that drFr/dr = 0.
```

Comment:
What is the formula in this inline comment? Is it $dF_r/dr=0$ or $d(rF_r)/dr=0$? In the second case I would maybe write
```suggestion
                                // Add radial scale so that d(rF_r)/dr = 0.
```